MoreLinq.Test/AggregateRightTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class AggregateRightTest
+    {
+        // Overload 1 Test
+
+        [Test]
+        public void AggregateRight1WithNullSequence()
+        {
+            Assert.ThrowsArgumentNullException("source", 
+                () => MoreEnumerable.AggregateRight<int>(null, (a, b) => a + b));
+        }
+
+        [Test]
+        public void AggregateRight1WithNullFunc()
+        {
+            Assert.ThrowsArgumentNullException("func", 
+                () => Enumerable.Range(1, 5).AggregateRight(null));
+        }
+
+        [Test]
+        public void AggregateRight1WithEmptySequence()
+        {
+            Assert.Throws<InvalidOperationException>(
+                () => new int[] { }.AggregateRight((a, b) => a + b));
```

Comment:
`new int[0]` is simpler and bears fewer tokens when parser is a human. 😄  Same applies to other places in the tests.